### PR TITLE
fix(compilerV2): Fixed inputs with constant arguments

### PR DESCRIPTION
**Pull request by Ark-kun**

### CODE REVIEW

File: sdk/python/kfp/dsl/_component_bridge.py [hunk 1]
```diff
@@ -582,12 +582,26 @@ def _resolve_ir_placeholders_v2(
...
 
+  component_spec = copy.deepcopy(component_spec)
+
   if not component_spec.name:
     component_spec.name = _components._default_component_name
 
   # task.name is unique at this point.
   pipeline_task_spec.task_info.name = (dsl_utils.sanitize_task_name(task.name))
 
+  # IR only supports constant arguments for parameter inputs.
```
Comment (chensun) [inline sdk/python/kfp/dsl/_component_bridge.py:593]: The comment isn't 100% accurate, an input can reference a component input parameter which references PipelineJob. RuntimeConfig.parameters, and the value could be missing at compile time.

File: sdk/python/kfp/dsl/_component_bridge.py [hunk 2]
```diff
@@ -582,12 +582,26 @@ def _resolve_ir_placeholders_v2(
...
+  # Converting inputs that have constant arguments to parameter inputs.
+  inputs_with_constant_arguments = {
+    name
+    for name, argument in pipeline_task_spec.inputs.parameters.items()
+    if argument.runtime_value.HasField("constant_value")
+  }
+  for input_spec in component_spec.inputs or []:
+    if (input_spec.name in inputs_with_constant_arguments and
+        not type_utils.is_parameter_type(input_spec.type)):
```
Comment (chensun) [inline sdk/python/kfp/dsl/_component_bridge.py:602]: Looking at the existing implementation, I think this HasField("constant_value") would be true only when the input is already categorized as a parameter. 
Unless I missed something, there won't be a case when this if statement returns true.

File: sdk/python/kfp/dsl/_component_bridge.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/dsl/_component_bridge.py:593]: I guess my wording is ambiguous.
What I meant is: "constant argument" *implies*/*requires* "parameter input".
Not the other way around.

File: sdk/python/kfp/dsl/_component_bridge.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/kfp/dsl/_component_bridge.py:602]: You're right. I have not realized that the previous code block was only setting `constant_value` when input is already a parameter.
The proper solution is likely closer to https://github.com/kubeflow/pipelines/pull/5478

**Comment by rimolive** [comment]
Closing this PR. No activity for more than a year.

/close

**Pull request closed without merging**